### PR TITLE
Bookstor: Feature connect to api

### PR DESCRIPTION
**Pull request by gitbygaurav**
### Hey CR 😊

AidanHelloAidanGallagherHelloGIF

In this PR, I have :

- Used fetch for making API calls.
- Fetched book list from Bookstore API on Books page load.
  - Added an Action Creator, that returns a function.
  - Fetched list of books.
  - Dispatched another action (add an action type) that will pass API response to reducer and updates the state.
- Refactored add book and remove book features to persist changes in the server.
  - Changed existing Action Creators, so they return functions. 
  - Made changes by using API.
  - Dispatched an action (action type that existed before refactor) that will updates the state accordingly.

Thanks 🍻
Gaurav

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,36 +1,49 @@
...
+const FETCH_BOOKS = 'bookstore/books/FETCH_BOOKS';
+
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/CZDXQb2zMgHNbimDk0mO/books';
+
+const initialState = [];
+
+export const fetchBooks = (payload) => ({
+  type: FETCH_BOOKS,
   payload,
 });
```
Comment (Mihndim2020) [inline src/redux/books/books.js]: - [ ] According to the project [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md),  " `Use fetch or axios with createAsyncThunk for making API calls` ", please kindly refactor your code to use `createAsyncThunk` for creating thunk action creators. Something like so: 

```
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```
Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @gauravgangwarweb, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @gauravgangwarweb , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Good commit history ✔️
- Good PR description ✔️
- Your README file is professional ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gitbygaurav** [comment]
Hii @alexander16108 👋

Thanks for reviewing and approving my PR.
ThanksGIF

**Pull request merged**